components/movementsensor/merged/merged_test.go

```diff
@@ -0,0 +1,339 @@
+		Model: model,
+		API:   movementsensor.API,
+		ConvertedAttributes: &Config{
+			Orientation:        ori,
+			Position:           pos,
+			CompassHeading:     compass,
+			LinearVelocity:     linvel,
+			AngularVelocity:    angvel,
+			LinearAcceleration: linacc,
+		},
+	}
+}
+
+func setupMovementSensor(
+	name string, prop movementsensor.Properties,
+	errAcc, oriPropsErr bool,
+) movementsensor.MovementSensor {
+	ms := inject.NewMovementSensor(name)
+	ms.PropertiesFunc = func(ctx context.Context, extra map[string]interface{},
+	) (*movementsensor.Properties, error) {
+		if oriPropsErr && strings.Contains(name, "goodOri") {
+			return &prop, errTestProps
+		}
+		return &prop, nil
+	}
+	ms.AccuracyFunc = func(ctx context.Context, exta map[string]interface{}) (map[string]float32, error) {
+		if errAcc {
+			return nil, errTestAcrcy
+		}
+		return map[string]float32{"accuracy": 32}, nil
+	}
+
+	switch {
+	case strings.Contains(name, "Ori"):
+		ms.OrientationFunc = func(ctx context.Context, extra map[string]interface{}) (spatialmath.Orientation, error) {
+			return testOri, nil
+		}
+	case strings.Contains(name, "Pos"):
+		ms.PositionFunc = func(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+			return testGeoPoint, testAlt, nil
+		}
+	case strings.Contains(name, "Compass"):
+		ms.CompassHeadingFunc = func(ctx context.Context, extra map[string]interface{}) (float64, error) {
+			return testCompass, nil
+		}
+	case strings.Contains(name, "AngVel"):
+		ms.AngularVelocityFunc = func(ctx context.Context, extra map[string]interface{}) (spatialmath.AngularVelocity, error) {
+			return testAngVel, nil
+		}
+	case strings.Contains(name, "LinVel"):
+		ms.LinearVelocityFunc = func(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+			return testLinVel, nil
+		}
+	case strings.Contains(name, "LinAcc"):
+		ms.LinearAccelerationFunc = func(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+			return testLinAcc, nil
+		}
+	}
+
+	return ms
+}
+
+func setupDependencies(t *testing.T,
+	depswithProps map[string]movementsensor.Properties,
+	errAcc, errProps bool,
+) resource.Dependencies {
+	t.Helper()
+	result := make(resource.Dependencies)
+	for name, prop := range depswithProps {
+		ms := setupMovementSensor(name, prop, errAcc, errProps)
+		result[movementsensor.Named(name)] = ms
+	}
+	return result
+}
+
+func TestValidate(t *testing.T) {
+	// doesn't pass validate
+	conf := setUpCfg(
+		oriSensors, []string{ /*pos*/ }, []string{ /*compass*/ },
+		[]string{ /*linvel*/ }, []string{ /*angvel*/ }, []string{ /*linacc*/ })
+	implicits, err := conf.Validate("somepath", movementsensor.API.Type.Name)
+	test.That(t, err, test.ShouldBeError, errMoreSensorNeeded)
+	test.That(t, implicits, test.ShouldBeNil)
+
+	// doesn't pass configuration
+	conf = setUpCfg(
+		oriSensors, []string{ /*pos*/ }, []string{ /*compass*/ },
+		linVelSensors, []string{ /*angvel*/ }, []string{ /*linacc*/ })
```

Comment:
Should I remove the inline comments?

Reply:
go fmt won't let me leave them on the left for some reason, kept them with a new test empty properties variable. Maybe overkill, but I like it.